src/set_relation/isl_str_manipulation.cc

```diff
@@ -0,0 +1,291 @@
+    for( uint i = 0; tv2[nonSpSt] == ' ' && i < tv2.length(); i++, nonSpSt++ );
+    for( uint i = 0; tv2[nonSpFi] == ' ' && i < tv2.length(); i++, nonSpFi-- );
+    tv2 = tv2.substr(nonSpSt,nonSpFi-nonSpSt+1);
+
+    if ( tv1 == tv2 ){
+        return true;
+    }
+
+    return false;
+}
+/*! This function constructs equality constraints between Tuple Varialbes
+**  that are replaced eachother by ISL. To do this, it takes in two
+**  Tuple declaration, extracts their Tuple Variables, then creates
+**  equalities for those that are replaced:
+**      origTupDecl: [i1, i2] -> [1, i4]
+**      islTupDecl : [col_tv1_, i2] -> [1, i2]
+**      output:      (i1) - (col_tv1_) = 0 and (i4) - (i2) = 0
+**  Note: noFirstAnd determines whether we should put "and" at the beginning of
+**  the output string, which depends on original constraints being empty or not. 
+*/
+std::string missingEqs(std::string origTupDecl, std::string islTupDecl,
+                              int inArity, int outArity, bool noFirstAnd){
+    std::string eqs("");
+    int arity = inArity + outArity;
+    bool firstEq = true;
+ 
+    std::string* origTupVars = tupVarsExtract(origTupDecl, inArity, outArity);
+    std::string* islTupVars = tupVarsExtract(islTupDecl, inArity, outArity);
+
+    for(int i = 0 ; i < arity ; i++){
+        if( ! tvEqCheck(origTupVars[i] , islTupVars[i]) ){
+            if(noFirstAnd && firstEq){
+                eqs +=  " (" + origTupVars[i] +
+                        ") - (" + islTupVars[i] + ") = 0";
+                firstEq = false;
+            } else {
+                eqs +=  " and (" + origTupVars[i] +
+                        ") - (" + islTupVars[i] + ") = 0";
+            }
+        
+        }
+    }
+
+    delete[] origTupVars;
+    delete[] islTupVars;
+
+    return eqs;
+}
+/*! This function takes in a Set or Relation string and returns different 
+**  parts of it in a srParts structure.
+*/
+srParts* getPartsFromStr(std::string str){
+    srParts* parts = new srParts();
+    int l1,l2,l3;
+
+    l1 = str.find_first_of('{', 0);
+    l2 = str.find_first_of(':', l1);
+    if( l2 < 0 ){
+        parts->symVars =     str.substr( 0 , l1-0 );
+        l3 = str.find_first_of('}', l1);
+        parts->tupDecl =     str.substr( l1+1 , l3-(l1+1)-1 );
+        parts->constraints = "";
+        return parts;
+    }
+    l3 = str.find_first_of('}', l2);
+
+    parts->symVars =     str.substr( 0 , l1-0 );
+    parts->tupDecl =     str.substr( l1+1 , l2-(l1+1) );
+    parts->constraints = str.substr( l2+1 , l3-(l2+1) );
+
+    return parts;
+}
+/*! The main function that restores uninted changes ISL library apply to
+**  Tuple Declaration because of the equality constraints. ISL library replaces
+**  tuple variables with their equal expression if one exists in the constraints:
+**    input to isl   : [n] -> { [i] : i = n and i < 10 }
+**    output from isl: { [n] : n < 10 }
+**  This function replaces Tuple Declaration from ISL string with original one,
```

Comment:
Check if [n] -> is still in output from ISL.
Put in what the output from this routine is going to be.